doc/sphinx/using/libraries/index.rst

```diff
@@ -4,6 +4,26 @@
 Libraries and plugins
 =====================
 
+:gdef:`Libraries <library>` contain only compiled Coq scripts while
+:gdef:`plugins <plugin>` also include compiled OCaml code that can change
+the behavior of Coq.  Both are :term:`packages <package>`.
+While users configure and load them identically, there are a few differences
+to consider:
+
+- Nearly all plugins add things to the vernacular language that could not be
+  added otherwise (modulo MetaCoq), so there are likely new top-level commands or
+  tactics.
+- Plugins can do almost anything to Coq, including expanding the trusted code
+  base, so they can't be fully trusted.  If necessary, you can eliminate
+  this risk by running :ref:`coqchk` on compiled scripts that load plugins.
+  (`coqchk` doesn't load plugins, so they won't be part of trusted code base.)
```

Comment:
Is this really correct?  You can modify the behavior of the kernel without editing the files that comprise it, e.g. you can plugin in additional code into the kernel?

Reply:
I don't think so, but it's definitely possible to completely bamboozle the kernel by mutating values after sending them to it. For instance it would be pretty easy to write a plugin which finds a constant whose type is `id True` (reduces to True) and modifies it to `id False`, because application nodes use ocaml's mutable arrays.